app/models/service.rb

```diff
@@ -0,0 +1,35 @@
+class Service < Aggregate
+  MAXIMUM_COMPONENTS_FOR_SERVICE = 2
+  SERVICE_PROVIDERS = /^VSP|SP/.freeze
+
+  has_and_belongs_to_many :components, before_add: %i[maximum_components_for_service
+                                                      existing_component_type
+                                                      single_service_provider]
+
+private
+
+  def maximum_components_for_service(_component)
```

Comment:
we don't need to pass the component in
```suggestion
  def maximum_components_for_service
```

Reply:
oh it's passed automagically